pkg/controllers/vdb/imageversion_reconciler_test.go

```diff
@@ -88,12 +91,39 @@ vertica(v11.1.0) built by @re-docker2 from tag@releases/VER_10_1_RELEASE_BUILD_1
 				},
 			},
 		}
-		r := MakeVersionReconciler(vdbRec, logger, vdb, fpr, &pfacts, true)
+		r := MakeImageVersionReconciler(vdbRec, logger, vdb, fpr, &pfacts, true)
 		Expect(r.Reconcile(ctx, &ctrl.Request{})).Should(Equal(ctrl.Result{Requeue: true}))
 
 		// Ensure we didn't update the vdb
 		fetchVdb := &vapi.VerticaDB{}
 		Expect(k8sClient.Get(ctx, vapi.MakeVDBName(), fetchVdb)).Should(Succeed())
 		Expect(fetchVdb.ObjectMeta.Annotations[vmeta.VersionAnnotation]).Should(Equal(OrigVersion))
 	})
+
+	It("should fail the reconclier if we use wrong image", func() {
```

Comment:
The test should cover wrong image error for both admintools and vclusterops

Reply:
Thank for your comments, I'm trying to find a way to build an image using AT style to verity wrong image for vclusterops.

Reply:
unit tests and e2e tests are different. your should have unit test that cover both cases regardless of e2e test

Reply:
No image build needed. Only go code. In one case you have the vclusterops flag unset and a Podfact with admintoolsExists=false, on the other case  vclusterops flag set and a Podfact with admintoolsExists=true.

Reply:
Done!